### PR TITLE
Cluster resize gossip config

### PR DESCRIPTION
**Pull request by travisturner**
## Overview

Adds support for memberlist gossip configuration via pilosa.Config.

## Pull request checklist

- [x] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [x] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [ ] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [x] I have resolved any merge conflicts.
- [ ] I have included tests that cover my changes.
- [x] All new and existing tests pass.

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.

### CODE REVIEW

**Comment by travisturner** [comment]
@jaffee I applied and expanded on your work in #885 to the second commit in this PR.

File: gossip/gossip.go [hunk 1]
```diff
@@ -154,51 +157,82 @@ type gossipConfig struct {
...
+func NewGossipMemberSetWithTransport(name string, cfg *pilosa.Config, transport *Transport, server *pilosa.Server) (*GossipMemberSet, error) {
 
 	g := &GossipMemberSet{
 		LogOutput: server.LogOutput,
 	}
 
+	port := transport.Net.GetAutoBindPort()
+	host, _, err := net.SplitHostPort(cfg.Bind)
+	if err != nil {
+		return nil, err
```
Comment (jaffee) [inline gossip/gossip.go]: add context to errors (throughout this file)

File: gossip/gossip.go [hunk 2]
```diff
@@ -90,7 +90,7 @@ func (g *GossipMemberSet) Open() error {
 	g.memberlist, err = memberlist.Create(g.config.memberlistConfig)
 	g.mu.Unlock()
 	if err != nil {
-		return err
+		return fmt.Errorf("opening gossip member set: creating memberlist: %s", err)
```
Comment (jaffee) [inline gossip/gossip.go]: I've found that it's better to keep the context message local to the interior of the function. Whatever is calling this function should add the "opening gossip member set" bit, so this message would just be "creating memberlist:"

File: gossip/gossip.go [hunk 3]
```diff
@@ -166,14 +166,14 @@ func NewGossipMemberSetWithTransport(name string, cfg *pilosa.Config, transport
 	port := transport.Net.GetAutoBindPort()
 	host, _, err := net.SplitHostPort(cfg.Bind)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("new gossip member set with transport: split host port: %s", err)
```
Comment (jaffee) [inline gossip/gossip.go]: similarly, strip out the "new gossip member set with transport:"

File: gossip/gossip.go [hunk 4]
```diff
@@ -239,7 +239,7 @@ func NewGossipMemberSet(name string, cfg *pilosa.Config, server *pilosa.Server)
 func (g *GossipMemberSet) SendSync(pb proto.Message) error {
 	msg, err := pilosa.MarshalMessage(pb)
 	if err != nil {
-		return err
+		return fmt.Errorf("send sync: marshal message: %s", err)
```
Comment (jaffee) [inline gossip/gossip.go]: remove "send sync:"

File: gossip/gossip.go [hunk 5]
```diff
@@ -267,7 +267,7 @@ func (g *GossipMemberSet) SendSync(pb proto.Message) error {
 func (g *GossipMemberSet) SendAsync(pb proto.Message) error {
 	msg, err := pilosa.MarshalMessage(pb)
 	if err != nil {
-		return err
+		return fmt.Errorf("send async: marshal message: %s", err)
```
Comment (jaffee) [inline gossip/gossip.go]: remove "send async:"

File: gossip/gossip.go [hunk 6]
```diff
@@ -438,12 +438,12 @@ func NewTransport(host string, port int) (*Transport, error) {
 
 	net, err := newTransport(conf)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("new transport: new transport: %s", err)
```
Comment (jaffee) [inline gossip/gossip.go]: remove the initial "new transport:"

**Pull request merged**